lib/vrl/core/src/target.rs

```diff
@@ -39,34 +40,26 @@ pub trait Target: std::fmt::Debug + MetadataTarget + SecretTarget {
     /// When inserting into a coalesced path, the implementor is encouraged to
     /// insert into the right-most segment if none exists, but can return an
     /// error if needed.
-    fn target_insert(&mut self, path: &LookupBuf, value: Value) -> Result<(), String>;
+    fn target_insert(&mut self, path: &TargetPath, value: Value) -> Result<(), String>;
 
     /// Get a value for a given path, or `None` if no value is found.
     ///
     /// See [`Target::insert`] for more details.
-    fn target_get(&self, path: &LookupBuf) -> Result<Option<&Value>, String>;
+    fn target_get(&self, path: &TargetPath) -> Result<Option<&Value>, String>;
 
     /// Get a mutable reference to the value for a given path, or `None` if no
     /// value is found.
     ///
     /// See [`Target::insert`] for more details.
-    fn target_get_mut(&mut self, path: &LookupBuf) -> Result<Option<&mut Value>, String>;
+    fn target_get_mut(&mut self, path: &TargetPath) -> Result<Option<&mut Value>, String>;
 
     /// Remove the given path from the object.
     ///
     /// Returns the removed object, if any.
     ///
     /// If `compact` is true, after deletion, if an empty object or array is
     /// left behind, it should be removed as well, cascading up to the root.
-    fn target_remove(&mut self, path: &LookupBuf, compact: bool) -> Result<Option<Value>, String>;
-}
-
-pub trait MetadataTarget {
-    fn get_metadata(&self, _path: &LookupBuf) -> Result<Option<Value>, String>;
-
-    fn set_metadata(&mut self, _path: &LookupBuf, _value: Value) -> Result<(), String>;
-
-    fn remove_metadata(&mut self, _path: &LookupBuf) -> Result<(), String>;
+    fn target_remove(&mut self, path: &TargetPath, compact: bool) -> Result<Option<Value>, String>;
```

Comment:
Yes, I actually want to change _all_ of the methods on the `Target` trait to be infallible, but since it's mostly unrelated to these changes, that will come in a later PR.
I already created an issue for it: https://github.com/vectordotdev/vector/issues/13920